### PR TITLE
Collin lib 12

### PR DESCRIPTION
**Pull request by paradigm-lab**
Refactor: #12

### CODE REVIEW

**Comment by paradigm-lab** [comment]
I have got a problem in the `MobilePayment` interface with there impl classes where by the classes that need to implement the method they need to implement all method even thought they do not need to implement the method. 
Asking for a solution from this problem.

**Comment by mattjquinn** [comment]
> I have got a problem in the MobilePayment interface with there impl classes where by the classes that need to implement the method they need to implement all method even thought they do not need to implement the method.
Asking for a solution from this problem.

Sijakuelewa. I ran `./gradlew testClasses` and everything compiled.

File: mpesa-mobile-payment-impl/src/main/java/com/kwawingu/payments/EncryptApiKey.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import java.security.InvalidKeyException;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import java.util.Optional;
+
+public class EncryptApiKey implements MobilePayment {
```
Comment (paradigm-lab) [inline mpesa-mobile-payment-impl/src/main/java/com/kwawingu/payments/EncryptApiKey.java:21]: This class implements `MobilePayment` and it needs to override all the methods from the interface and I don't want to implement the other method.

File: mpesa-mobile-payment-impl/src/main/java/com/kwawingu/payments/EncryptApiKey.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            rsaCipher.init(Cipher.ENCRYPT_MODE, pubKey);
+            byte[] encryptedBytes = rsaCipher.doFinal(apiKey.getBytes());
+            return Base64.getEncoder().encodeToString(encryptedBytes);
+        } catch (NoSuchAlgorithmException | InvalidKeySpecException | NoSuchPaddingException | InvalidKeyException |
+                 IllegalBlockSizeException | BadPaddingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
```
Comment (paradigm-lab) [inline mpesa-mobile-payment-impl/src/main/java/com/kwawingu/payments/EncryptApiKey.java:48]: When the client creates an object of this class he/she will be able to see the method which does return an empty option. I thought of making the overridden method as private but I thought it's not a great solution.

File: mpesa-mobile-payment-impl/src/main/java/com/kwawingu/payments/SessionKey.java [hunk 3]
```diff
@@ -1,77 +1,33 @@
...
-import java.security.spec.InvalidKeySpecException;
-import java.security.spec.X509EncodedKeySpec;
-import java.util.Base64;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
 
-public class MpesaConfig implements MobilePayment {
-    private static final Logger LOG = LoggerFactory.getLogger(MpesaConfig.class);
+public class SessionKey implements MobilePayment {
```
Comment (paradigm-lab) [inline mpesa-mobile-payment-impl/src/main/java/com/kwawingu/payments/SessionKey.java:18]: This class 2

File: mpesa-mobile-payment-impl/src/main/java/com/kwawingu/payments/EncryptApiKey.java (hunk 2 above)
Comment (mattjquinn) [reply mpesa-mobile-payment-impl/src/main/java/com/kwawingu/payments/EncryptApiKey.java:48]: If you don't want anyone to use the method, you have two main options:
* make the method `private` or `protected` or package-private.
* make the class implement an interface that doesn't contain the method definition, and only let people use that interface

Which method you choose depends on which design you want.

File: mpesa-mobile-payment-impl/src/main/java/com/kwawingu/payments/SessionKey.java (hunk 3 above)
Comment (mattjquinn) [reply mpesa-mobile-payment-impl/src/main/java/com/kwawingu/payments/SessionKey.java:18]: Same comment here as above; I don't think a key implements a payment; a key allows a payment to happen, but it isn't a payment.

**Pull request merged**